### ISSUE TITLE
Sometimes changing "Send as" identity, changes smtp sending server but keeps old "from: field".

### ISSUE THREAD

**Issue by ghost** [thunderbird/thunderbird-android#766]
This only happens about half the times, I start replying an email, change my identity from, say, hotmail to yahoo ( I have 3 ), and when I try to send, I receive a notification saying that the SMTP server ( In my example I would be the yahoo smtp ) refuses to send because I seeding from an non verified email. ( My from: field is still with the hotmail address. ).

**Comment by cketti** [comment]
Please provide detailed information about your setup (what accounts are set up, what identities, etc.) and step by step instructions on what you do when you encounter this.

**Comment by petruccigit** [comment]
I'm using cyanogen 12. It also happened before I flashed my phone, when I had android lolipop. It happens mostly when I receive an email, change from my default sending email, (hotmail) to another and try to reply. Not sure If it happens only when replying(If it also happens when writing a new email), but that what I mostly do. I exported my config to a xml, will it help if I send it to you? I only removed sensible information from the file. How should I send it?
Is there anything else that you need to know?
Thank you for your interest and quick response =).

**Comment by alanthehat** [comment]
I'm using Samsung J500FN running stock android 5.1.1, K-9 v5.008. 
This just happened to me & has happened before on other devices with other Android versions. 

Step-by-step -- 
Tapped 'send email' in another app, K-9 came up to handle email, 
Wrote message, 
Swapped to other apps several times, 
Selected different address in 'From' field, 
Changed message slightly, 
Tapped 'send' icon. 

Message was rejected by remote host with similar reason to OP.

I found message in 'outbox' of first (default) account, not in outbox of new account. 

Swapping to other apps part-way through composing seems to be an important feature to reproduce this bug.
